docs/rules/no-unused-expressions.md

```diff
@@ -31,6 +31,7 @@ This rule, in its default state, does not require any arguments. If you would li
 * `allowShortCircuit` set to `true` will allow you to use short circuit evaluations in your expressions (Default: `false`).
 * `allowTernary` set to `true` will enable you to use ternary operators in your expressions similarly to short circuit evaluations (Default: `false`).
 * `allowTaggedTemplates` set to `true` will enable you to use tagged template literals in your expressions (Default: `false`).
+* `enforceForJSX` set to `true` will flag unused JSX element expressions (Default: `false`).
```

Comment:
Sorry @duncanbeevers, I forgot about the examples of correct and incorrect code with each option below. Can you add those for this new option? That'd be a fine place to explain that people can enable this option if JSX has no side effects as in React. (Thanks @mdjermanovic for catching my mistake.)

Reply:
Sure thing. I considered adding those, but I thought they might be superfluous. I really don't want this PR to add a lot of noise, and I especially want to make sure anything public-facing gets explicit consideration.

Reply:
I've added examples and a little descriptive text about JSX pragmas. I'm still concerned about being too low-level, and really want something sweet, simple, and that clearly indicates to the 99.9999% of JSX users that they should turn this on.